tests/JustSaying.Extensions.DependencyInjection.AwsCore.Tests/ClientFactoryTests.cs

```diff
@@ -0,0 +1,106 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using JustSaying.AwsTools;
+using Amazon;
+
+
+[assembly: CollectionBehavior(DisableTestParallelization = true)]
+
+namespace TestProject1;
+
+
+public class ClientFactoryTests
+{
+    [Fact]
+    public void ShouldReadConfigFromFile()
+    {
+
+        var config = new ConfigurationBuilder()
+            .AddJsonFile("appsettings.json")
+            .Build();
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        var provider = serviceCollection.BuildServiceProvider();
+        var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
+        var snsClient = clientFactory.GetSnsClient(RegionEndpoint.EUWest1);
+        Assert.True(snsClient.Config.ServiceURL.Equals("http://test.test/") == true);
+        Assert.True(snsClient.Config.UseHttp);
+    }
```

Comment:
Break the tests up into arrange, act and assert blocks. Otherwise it's just a wall of code.